app/backend/approaches/tabulardataassistant.py

```diff
@@ -0,0 +1,235 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import base64
+import os
+import glob
+import warnings
+import re
+import pandas as pd
+from langchain.chat_models import ChatOpenAI
+from langchain_experimental.agents.agent_toolkits import create_pandas_dataframe_agent
+from langchain.agents.agent_types import AgentType
+from langchain.chat_models import AzureChatOpenAI
+from langchain.agents import load_tools
+import matplotlib.pyplot as plt
+
+warnings.filterwarnings('ignore')
+from dotenv import load_dotenv
+
+# Initialize session state
+# if 'show_images' not in st.session_state:
+#     st.session_state.show_images = False
+
+#--------------------------------------------------------------------------
+#variables needed for testing
+OPENAI_API_TYPE = "azure"
+OPENAI_API_VERSION = "2023-06-01-preview"
+OPENAI_API_BASE = " "
+OPENAI_API_KEY = " "
+OPENAI_DEPLOYMENT_NAME = " "
+MODEL_NAME = " "
+AZURE_OPENAI_ENDPOINT = ' '
+AZURE_OPENAI_SERVICE_KEY = ' '
+
+os.environ["OPENAI_API_TYPE"] = OPENAI_API_TYPE
+os.environ["OPENAI_API_VERSION"] = OPENAI_API_VERSION
+#os.environ["OPENAI_API_BASE"] = OPENAI_API_BASE
+#os.environ["AZURE_OPENAI_ENDPOINT"] = AZURE_OPENAI_ENDPOINT
+#os.environ["AZURE_OPENAI_SERVICE_KEY"] = AZURE_OPENAI_SERVICE_KEY
+#os.environ["OPENAI_DEPLOYMENT_NAME"] = OPENAI_DEPLOYMENT_NAME
+
+load_dotenv()
+
+#Environment variables when integrated into the app
+#_________________________________________________________________________
+
+# # Access environment variables
+# azure_openai_service_key = os.getenv("AZURE_OPENAI_SERVICE_KEY")
+# azure_openai_service = os.getenv("AZURE_OPENAI_SERVICE")
+azure_openai_chatgpt_deployment = os.getenv("AZURE_OPENAI_CHATGPT_DEPLOYMENT")
+
+# openai.api_key = azure_openai_service_key
+# openai.api_base = f"https://{azure_openai_service}.openai.azure.com/"
+deployment_name = azure_openai_chatgpt_deployment
+OPENAI_DEPLOYMENT_NAME = deployment_name
+
+# openai.api_type = "azure"
+# openai.api_version = "2023-06-01-preview"
+# 
+#______________________________________________________________________________________
+
+
+# Page title
+
+
+# uploaded_file = st.file_uploader('Upload a CSV file', type=['csv'])
+dffinal = None
+pdagent = None
+agent_imgs = []
+
+def refreshagent():
+    global pdagent
+    pdagent = None
+
+
+def save_chart(query):
```

Comment:
save_chart function is not being used in this code. There are two commented-out sections where it appears to have been intended for use but as they are commented out. Is this the intention? probably good idea to clean it up if it is not being used.

Reply:
removed